views/index.html

```diff
@@ -150,24 +159,48 @@
         success: false,
         results: [],
       },
+      genericError: null,
+    },
+    created: function() {
+      const url = new URL(window.location);
+      this.formData.host = url.searchParams.get('host');
+      this.formData.ingressProxyPath = url.searchParams.get('ingressProxyPath');
+      this.formData.cdnProxyPath = url.searchParams.get('cdnProxyPath');
+      this.formData.trafficName = url.searchParams.get('trafficName');
+      this.formData.integrationVersion = url.searchParams.get('integrationVersion');
     },
     methods: {
-      handleSubmit() {
+      async handleSubmit() {
         this.testResult.status = 'pending';
         this.testResult.results = [];
-        const vm = this;
-        fetch('/api/test/run-tests', {
+        this.genericError = null;
+        const response = await fetch('/api/test/run-tests', {
           method: 'POST',
           headers: {
             'Content-Type': 'application/json',
           },
           body: JSON.stringify({ ...this.formData }),
-        }).then(res => res.json()).then(res => {
-          vm.testResult.success = res.results.every(t => t.passed === true);
-          vm.testResult.results = res.results.map((item, index) => ({ ...item, id: `element${index}` }));
-        }).finally(() => {
-          vm.testResult.status = 'finished';
         });
+
+        if (!response.ok) {
+          if (response.status === 500) {
```

Comment:
What if we will get other status code?

Reply:
Ah Yes, I'll handle 400 error as well, I'll create a task and handle it in different PR 👍